src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -3071,11 +3068,15 @@ func LaunchCustomQemu(ctx context.Context, path string, params []string, fds []*
 	cmd.Stderr = &stderr
 	logger.Infof("launching %s with: %v", path, params)
 
-	err := cmd.Run()
+	err := cmd.Start()
 	if err != nil {
 		logger.Errorf("Unable to launch %s: %v", path, err)
-		errStr = stderr.String()
-		logger.Errorf("%s", errStr)
+		return 0, err
 	}
-	return errStr, err
+
+	go func() {
```

Comment:
The commit description indicates that you intend to _defer_, so I wonder if the intent was to have a `defer func()` here. If not, I do not understand how that is supposed to work, and a comment would be welcome.

Reply:
> The commit description indicates that you intentionally defer the `Wait` to a goroutine. But since there is no channel or anything to communicate with that goroutine, I do not understand what you expect the `Wait()` to actually do for you?
> 

`Wait` must be called for reaping purpose, especially in the case QEMU isn't a daemon.

> Maybe it's a hook for a later commit? If so it would be nice for the commit message to explain. If there is another reason for that `Wait()`, I must say it is mysterious to me.
> 
> Update: It is indeed used for logging in a later commit.

Yeah. The idea is that `Wait` also closes the `stderr` pipe that will be added for logging. As explained at https://go.dev/src/os/exec/exec.go#L780 , `It is thus incorrect to call Wait before all reads from the pipe have completed.`

I can document that.